shell/common/shell.h

```diff
@@ -406,9 +406,12 @@ class Shell final : public PlatformView::Delegate,
   mutable std::mutex time_recorder_mutex_;
   std::optional<fml::TimePoint> latest_frame_target_time_;
   std::unique_ptr<PlatformView> platform_view_;  // on platform task runner
-  std::unique_ptr<Engine> engine_;               // on UI task runner
-  std::unique_ptr<Rasterizer> rasterizer_;       // on raster task runner
-  std::unique_ptr<ShellIOManager> io_manager_;   // on IO task runner
+  /// A mutex that blocks the platform_view_ from being deleted while it is
+  /// handling FFI platform messages.
+  std::mutex platform_view_mutex_;
```

Comment:
Is this lock needed?

HandleFfiPlatformMessage should be like other engine APIs that are called synchronously from Dart.  The Shell destructor will ensure that the Dart runtime is shut down before the platform view is deleted.

Reply:
During normal execution it isn't needed since like you said the callers originate from Dart.  The main reason I did it is to protect us when we are doing tests.

Reply:
Can the tests match the usage patterns of the full system?

I'd prefer to avoid adding more complexity to the shell's threading model.

Reply:
The tests I've written are all integration tests and don't need this level of safety.  I can probably write all the tests like that.  It should be fine but I'm a bit wary of relying on convention for something I can't enforce with some sort of assert.  Any idea how I could assert the caller is Dart?

I could clean this up if we introduced a concept that was a mutex+unique_ptr.  That could clean up the code so that all the old code would look the same except it would just be a different type.